src/dotnet-core-uninstall/Shared/Utils/Regexes.cs

```diff
@@ -30,7 +30,7 @@ internal static class Regexes
             $@"(?<{ArchGroupName}>\-?x64|x86)");
 
         private static readonly Regex _previewVersionSdkDisplayNameRegex = new Regex(
-            $@"(?<{PreviewGroupName}>\s?\-\s?((preview|alpha)\.?{_previewVersionNumberRegex.ToString()}|rc{_rcVersionNumberRegex.ToString()}))");
+            $@"(?<{PreviewGroupName}>.*((preview|alpha)[\.\-\s]?{_previewVersionNumberRegex.ToString()}|rc{_rcVersionNumberRegex.ToString()}))");
```

Comment:
I'm not sure I understand this regex properly. It seems like it accepts `<preview>` followed by anything followed by preview then more stuff? I don't think I've seen `<preview>-preview`.

Reply:
Here, `<preview>` serves as a named group, so it doesn't capture the text _per se_. 
![image](https://github.com/user-attachments/assets/0b262c73-ecda-4515-803e-90f5af4c817a)